Minor log changes, docker binary changes

will-chan opened the pull request:
- Added logrus log support
- Changed docker binary from "docker" to now standard "docker-machine".  This will also be applicable to MacOS developers as well.
- Added stderr support when logging docker machine output.
- Added Machine Storage Path support that the new Docker machine client requires.
- Tested against latest Docker Machine Beta (https://github.com/docker/machine/releases/tag/v0.1.0)
- Added additional logging to track go worker routine execution by specifying resourceId that is currently being worked on.

will-chan committed:
Added logrus support, docker machine stderr logs, and rename base docker binary to docker-machine

will-chan committed:
Added support for logrus, stderr printout for machine, and renaming
docker binary executable to docker-machine

cjellick commented on events/listener.go:
```diff
@@ -101,7 +101,7 @@ func (router *EventRouter) Start(ready chan<- bool) (err error) {
 			go worker.DoWork(line, handlers, router.apiClient, workers)
 		default:
-			log.Printf("No workers available dropping event.")
+			log.Debug("No workers available dropping event.")
```
I'd like this to be Info level. If we consistently don't have workers available, we need to know it so that we can tune the number of workers.

cjellick commented on events/listener.go:
```diff
@@ -124,14 +124,20 @@ func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler,
-		log.Printf("Error unmarshalling event: %v", err)
+		log.WithFields(log.Fields{
+			"Err": err,
+		}).Warn("Error unmarshalling event")
```
I'd make this error level

cjellick replied:
Looking again, I stand by making this error level. Failure to get the json payload means cattle and go-machine-service are failing to communicate as expected and it should be looked into.

cjellick commented on events/listener.go:
```diff
@@ -139,11 +145,15 @@ func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler,
+				"Event ID":    event.Id,
+				"Resource ID": event.ResourceId,
+				"Err":         err,
+			}).Warn("Error processing event")
```
I'd make this error level

cjellick commented on events/listener.go:
```diff
@@ -139,11 +145,15 @@ func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler,
 		}
 	} else {
-		log.Printf("No handler registered for event %v", event.Name)
+		log.Debug("No handler registered for event %v", event.Name)
```
It's actually a pretty odd case that we'd hit this logic and it should show up in non-debug logs if it happens. I'm think this should be warn level.

cjellick commented:
There's one bigger conceptual change I'd like to make to the logging:

In [listener.go's worker's doWork method](https://github.com/rancherio/go-machine-service/blob/master/events/listener.go#L120), log the full raw event json instead of just logging the event name.

Then at the beginning of each handler ([like here](https://github.com/rancherio/go-machine-service/blob/master/handlers/bootstrap.go#L25)), instead of logging the full event object, just log the event id and resource id.

We should not log the `event` object/struct anywhere. It's unreadable and somewhat useless in the logs.

cjellick commented:
Generally, for field names, dont use two words, use camel case. So, for example, instead of `"Event Name"`, do `"EventName"`.

cjellick commented on handlers/create.go:
```diff
@@ -146,6 +161,6 @@ func buildMachineCreateCmd(machine *client.Machine) ([]string, error) {
 	cmd = append(cmd, machine.Name)
 
-	log.Printf("Cmd slice: %v", cmd)
+	log.Infof("Cmd slice: %v", cmd)
```
I'm thinking just delete this log statement all together.

cjellick commented on handlers/remove.go:
```diff
@@ -48,8 +54,10 @@ func PurgeMachine(event *events.Event, apiClient *client.RancherClient) error {
 
-	log.Printf("Done purging machine. ResourceId: %v. ExternalId: %v.", event.ResourceId,
-		machine.ExternalId)
+	log.WithFields(log.Fields{
```
Can delete this log statement all together. We have the log statement on the entry to this function.

cjellick commented on main.go:
```diff
@@ -1,14 +1,21 @@
 func main() {
-	log.Println("Beginning go-machine-service...")
+	// Process options
+	for _, f := range os.Args {
```
Two things:
1. This is good enough for now, but eventually we'll probably want to support --log-level=[debug|info|warn|error]. Would it be much  more effort for you to change it to that?
2. Even if you don't do number 1, can you extract this to a function called `setLogLevel` or something like that?

cjellick commented:
@will-chan I'm done reviewing. Ping me if you have any questions.

will-chan committed:
Added additional logging to indicate execution of worker routines per
machine call.  Also, added supported new docker machine directory
changes from machine_dir to machine_storage_path

cjellick commented on events/listener.go:
```diff
@@ -139,11 +145,17 @@ func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler,
+				"EventId":    event.Id,
+				"ResourceId": event.ResourceId,
+				"Err":        err,
+			}).Warn("Error processing event")
```
This was the other place I suggested error level but github hid the original comment when you changed the field names to camel case.

I stand by making this an error level log as well. This error represents the errors raised by the handlers. If a handler is raising an error, that means rancher is failing to deploy a machine. I think it is reasonable for us to log this as an error and that Ops guys would want to see if machines were failing to deploy.

That said, the other thing I ultimately want to do here is update the transitioningMessage on machine api resource with the error so that the user can know how/why the machine errored out.

cjellick commented:
@will-chan I think I'm going to merge this as-is. The remaining unaddressed comments are small, not very important, and can be addressed in future iterations. The only exception is the logging of the full json event payload when an event is received; I'll open a separate issue for that.
